util/releasenotes.py

```diff
@@ -53,59 +93,118 @@ def contents(self):
-                        self._component_name = match.group("name")
+                        self._component.name = match.group("name")
                     continue
 
-                # find latest version
-                if not self._latest_version:
-                    match = re.search(ReleaseNotes._VERSION_PATTERN, line)
-                    if match:
-                        self._latest_version = match.group("version")
-                        self._latest_descriptor = match.group("descriptor")
-                        self._latest_date = match.group("date")
+                # parse version lines
+                version_match = re.search(ReleaseNotes._VERSION_PATTERN, line)
+                if version_match:
+                    descriptor = version_match.group("descriptor")
+                    number = version_match.group("version")
+                    date = version_match.group("date")
+                    version = Version(descriptor, number, date)
+                    self._component.versions.append(version)
+                    self._flush_details(details_buffer, last_object)
+                    details_buffer = []
+                    last_object = version
                     continue
 
-                # finished parsing what's needed
-                break
+                # parse note version lines
+                note_match = re.search(ReleaseNotes._NOTE_PATTERN, line)
+                if note_match:
+                    title = line.rstrip().lstrip()
+                    ticket = note_match.group("ticket")
+                    note = Note(title, ticket)
+                    self._component.versions[-1].notes.append(note)
+                    self._flush_details(details_buffer, last_object)
+                    details_buffer = []
+                    last_object = note
+                    continue
 
+                # not the component name, nor a version, nor a note - treat as a detail line
+                details_buffer.append(line)
+            self._flush_details(details_buffer, last_object)
             self._parsed = True
+
+        # if dependencies not already set and there are dependencies to add...
+        if not self._dependencies and self.dependencies():
+            self._component.dependencies = self._dependencies
+
         return self._parsed
 
-    def component_name(self):
+    @staticmethod
+    def _flush_details(details_buffer, object_with_details):
+        # strip extraneous whitespace at beginning and end of combined string
+        details = "".join(details_buffer).rstrip().lstrip() if details_buffer else None
+        if details:
+            object_with_details.details = details
+
+    def get_component_name(self):
         """
         :return: the component name or None if no component name was found
         """
         self._parse()
-        return self._component_name
+        return self._component.name
 
-    def latest_version(self):
+    def get_latest_version(self):
```

Comment:
getters and setters are not "pythonic". If you need to have side effects consider using properties (c.f. https://www.geeksforgeeks.org/getter-and-setter-in-python/).

Reply:
I'm definitely interested in the self._parse() side affect and some of the defaulting logic too, so I'll use properties.